### PR TITLE
Disable pushprox client metrics by default

### PR DESCRIPTION
**Pull request by aiyengar2**
By default, we don't define a serviceMonitor anyways that collects pushprox **client** metrics (e.g. the metrics pushprox posts about itself) so this PR disables publishing those metrics by default.

While it may be a later RFE in order to actually support a ServiceMonitor capable of scraping those client metrics, this is being done to prevent PushProx from unnecessarily binding a hostNetwork port on the nodes to host these metrics.

Followup to https://github.com/rancher/charts/pull/1402.

Related Issue: https://github.com/rancher/rancher/issues/36140

### CODE REVIEW

File: packages/rancher-monitoring/rancher-pushprox/charts/values.yaml [hunk 1]
```diff
@@ -5,11 +5,6 @@
 # Default image containing both the proxy and the client was generated from the following Dockerfile
 # https://github.com/prometheus-community/PushProx/blob/eeadbe766641699129920ccfaaaa30a85c67fe81/Dockerfile#L1-L15
 
-# Note: if using a cloud provider, the nodes that contain the PushProxy client must allow the node(s) that contain(s)
```
Comment (aiyengar2) [inline packages/rancher-monitoring/rancher-pushprox/charts/values.yaml:8]: I don't think this note is accurate or necessary anymore. Might have been a remnant of the initial design of the chart.

**Review by dkeightley — DISMISSED** [review]
LGTM

**Review by brendarearden — DISMISSED** [review]
lgtm

**Review by dkeightley — DISMISSED** [review]

**Review by pennyscissors — APPROVED** [review]

**Comment by bkleiner** [comment]
It appears that this change does _not_ prevent the clients from opening a port, but makes it [fallback to the default port](https://github.com/prometheus-community/PushProx/blob/master/cmd/client/main.go#L50), exposing whichever pushprox client was started first.

**Pull request merged**